basictracer/span.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import absolute_import
+
+from threading import Lock
+import time
+import re
+
+from opentracing import Span
+from .context import Context
+from .util import generate_id
+
+class BasicSpan(Span):
+    """BasicSpan is a thread-safe implementation of opentracing.Span.
+    """
+
+    def __init__(self, tracer,
+            operation_name=None,
+            context=None,
+            tags=None,
+            sampled=False,
+            start_time=None):
+        super(BasicSpan, self).__init__(tracer)
+        self._tracer = tracer
+        self._lock = Lock()
+
+        self.operation_name = operation_name
+        self.start_time = start_time
+        self.context = context
+        self.tags = tags
+        self.duration = -1
+        self.logs = []
+
+    def set_operation_name(self, operation_name):
+        with self._lock:
+            self.operation_name = operation_name
+        return super(BasicSpan, self).set_operation_name(operation_name)
```

Comment:
per offline discussion, IMO this is not needed given the context we're in, but that's 6 vs half-dozen and I don't feel strongly